bin/ruby-build

```diff
@@ -1256,6 +1256,29 @@ build_package_ldflags_dirs() {
   done
 }
 
+build_package_enable_yjit() {
+  local rustc_installed rustc_ver
+
+  rustc_installed=""
+  if which rustc; then
+    rustc_ver=`rustc --version | cut -c7-10`
+    # Some kind of built-in bash comparison for dotted version strings would be awesome.
+    if [[ " ${rustc_ver} " == "1."[6789]* ]]; then
+      rustc_installed="1"
+    fi
+  fi
+
+  # If YJIT is explicitly specified, don't change anything
+  if [[ " ${RUBY_CONFIGURE_OPTS} " != *" --enable-yjit"* ]]; then
+    if [[ -z "${rustc_installed}" ]]; then
+      echo "rustc ${rustc_ver} is installed: enabling yjit by default"
+      package_option ruby configure --enable-yjit
+    else
+      echo "rustc 1.60+ is not installed; disabling yjit by default"
```

Comment:
I think that may be too much, notably because that page seems very opinionated (e.g., it does not even mention using a package manager, which has advantages, especially if one does not plan to develop in rust but just build rust software).
And in general if people want yjit I'm confident they can find how to install rustc, OTOH for users which do not mind about YJIT/do not well about installing extra things then it seems better to not lead to yet another step of installing packages which could fail, especially a custom one like this with curl.

I'd phrase it like:
```
rustc 1.60+ is not installed, YJIT will not be built; if you want to use YJIT, install rustc 1.60+ and rerun this command.
```